### ISSUE TITLE
Helm: Support for OCI-based registries

### ISSUE THREAD

**Issue by goenning** [aptakube/aptakube#313]
### What problem are you trying to solve? Do you a suggestion on how to solve it?

ref: https://helm.sh/docs/topics/registries/

**Comment by corvis** [comment]
That would be very helpful for us

**Comment by goenning** [comment]
Does it currently throw an error? Do you have a screenshot of the issue?

**Comment by corvis** [comment]
When I open "Upgrade Release" dialog for the helm chart which points OCI registry I see no versions in "Target" dropdown so Upgrade function doesn't work (see below):

<img width="700" height="576" alt="Image" src="https://github.com/user-attachments/assets/c26f6b91-e3eb-4350-8178-cb300eb2af42" />

At the same time when I do the same for other release the dropdown is populated:

<img width="704" height="515" alt="Image" src="https://github.com/user-attachments/assets/18561782-42cc-42c4-bef5-1d40ff8031dd" />

So that makes me think the issue is in OCI...

**Comment by goenning** [comment]
oh, it seems like OCI registries don't support `helm search`, as I use this command `helm search repo keda --versions -o json` to find the list of versions. I need to find an alternative...

**Comment by corvis** [comment]
Thanks for the prompt reply. To me, if it is impossible to run a search, just replacing drop-down with a text field pre-populated with current version will be a good workaround. It is much better than fully nonfunctional feature.

**Comment by smeng9** [comment]
Hi @goenning 

Any updates on this? Some of the helm charts have already migrated to oci format and deprecated http format. Without version selection it becomes really hard to use......

We could consider calling external CLI commands like
```
oras repo tags quay.io/jetstack/charts/cert-manager
```
to find the list of versions

**Comment by kedvall** [comment]
Hi, I just wanted to confirm that using `oras` seems to work well for me too. It seems this could be a suitable solution.

Integrating this would be a huge help as all the charts we use have migrated to recommending OCI installs. Thanks!

**Comment by goenning** [comment]
hey all @corvis @kedvall @smeng9 I've been playing with oras and it seems like it can be a good fit.

I'm new to helm charts in OCI, and I'm struggling to find two things:
- how to identify if a helm release was installed via OCI or Repo
- which OCI registry was used to install the chart

```
helm install \
  cert-manager oci://quay.io/jetstack/charts/cert-manager \
  --version v1.20.0 \
  --namespace cert-manager \
  --create-namespace \
  --set crds.enabled=true
```

 works fine, but it seems like that `oci://quay.io/jetstack/charts/cert-manager` is not stored anywhere:

```
helm get all cert-manager -n cert-manager | grep charts
```

returns nothing, any ideas?

**Comment by smeng9** [comment]
Blocked by https://github.com/helm/helm/pull/31792

**Comment by corvis** [comment]
Is there any way to unblock upgrade process when it is impossible to extract meta data about versions so at least updating values will work when you do not want to change version (which is pretty common case at least for me).

**Comment by goenning** [comment]
@corvis the issue is that `helm upgrade` always requires the URL of the chart, be it HTTP or OCI based. 

Unless I'm missing another command that does it? How do you usually update the values?

**Comment by goenning** [comment]
My bad, it seems like it's only OCI that requires the URL, @corvis let me see what I can do, but I think that's possible

**Comment by corvis** [comment]
He @goenning, thank you for the quick reply. Here is the my example script i use to update `image.tag` the same is used to update env vars:

```
helm upgrade \
    --atomic \
    --timeout 900s \
    --version $HELM_CHART_VERSION \
    --values - \
    --set image.tag="1.2.3" \
    $HELM_RELEASE_NAME \
    oci://REDACTED.azurecr.io/$HELM_CHART_NAME
```

If I got you right the problem is to get URL, is this correct?

Also I just found a script we use for fetching versions (in a different context but still):
```
versions=$(curl  -s -u $HELM_REPOSITORY_USER:$HELM_REPOSITORY_PASSWORD "https://REDACTED.azurecr.io/v2/$HELM_CHART_NAME/tags/list" | jq -r ".tags[]")
```
It seems like at least Azure Container registry delivers metadata via http, not sure if this is a standard but if it is maybe aptakube can try to query HTTP for metadata when detects OCI URL?

**Comment by goenning** [comment]
Yes, Helm doesn't store that URL anywhere, not even if was installed using a OCI URL, so there's no way for us to know if the chart is in Azure Container Registry, or AWS Registry, etc 

I'm watching progress on https://github.com/helm/helm/pull/31792 to see if that'll help

**Comment by smeng9** [comment]
Hi @goenning 

Do you have any helm maintainer that you can reach out to so we can request a review from them?
https://github.com/helm/helm/pull/31792 is a feature wanted by a lot of community members.

**Comment by goenning** [comment]
I don't, unfortunately :(
